lib/auth/backends/in_memory/Backend.ts

```diff
@@ -61,7 +70,15 @@ class InMemoryBackend extends BaseBackend {
             // @ts-ignore
             IAMdisplayName: entity.IAMdisplayName,
         };
-        const vaultReturnObject = this.formatResponse(userInfoToSend);
+        const vaultReturnObject = this.formatResponse(
+            userInfoToSend,
+            [],
+            {
+                account: entity.canonicalID,
+                // account quota is not supported in this backend
+                accountQuota: 0,
+            }
```

Comment:
nit: would it not be simpler to just add a `cannonicalId` or `entity` parameter to `formatResponse`, and let that function handle the formatting of nested types and filling of missed fields just once?

```
function _formatResponse(userInfoToSend: AuthInfoType, account: string): { message: { body: AuthV4Results } } {
    return {
        message: {
            body: {
                userInfo: userInfoToSend,
                authorizationResults: [],
                accountQuota: {
                    account,
                    // account quota is not supported in this backend
                    accountQuota: 0,
                },
            },
        },
    };
}
```

```suggestion
const vaultReturnObject = this.formatResponse(userInfoToSent, entity.canonicalID);
```